### PR TITLE
validate namespace before setup and teardown

### PR DESCRIPTION
**Pull request by lsm5**
Signed-off-by: Lokesh Mandvekar <lsm5@fedoraproject.org>

### CODE REVIEW

**Comment by lsm5** [comment]
@flouthoc your suggestion worked. PTAL and let me know.

File: src/commands/setup.rs [hunk 1]
```diff
@@ -20,6 +20,11 @@ impl Setup {
     }
 
     pub fn exec(&self, input_file: String) {
+        assert!(
+            network::validation::ns_checks(&input_file),
```
Comment (flouthoc) [inline src/commands/setup.rs:24]: This has to validate network namespace so field should point to network namespace.
```suggestion
            network::validation::ns_checks(&self.network_namespace_path),
```

File: src/commands/teardown.rs [hunk 2]
```diff
@@ -18,6 +18,11 @@ impl Teardown {
     }
 
     pub fn exec(&self, input_file: String) {
+        assert!(
+            network::validation::ns_checks(&input_file),
```
Comment (flouthoc) [inline src/commands/teardown.rs:22]: same as above
```suggestion
            network::validation::ns_checks(&self.network_namespace_path),
```

File: src/test/test.rs [hunk 3]
```diff
@@ -36,4 +36,13 @@ mod tests {
             Err(e) => panic!("{}", e),
         }
     }
+
+    // Test commands::setup::ns_checks works correctly
+    #[test]
+    fn test_ns_checks() {
+        assert_eq!(
+            network::validation::ns_checks("src/test/config/setupopts.test.json"),
```
Comment (flouthoc) [inline src/test/test.rs:44]: This is fine since we dont have a network namespace for testing as of now.

**Comment by flouthoc** [comment]
@lsm5 LGTM just few nits

**Comment by lsm5** [comment]
> @lsm5 LGTM just few nits

@flouthoc thanks. Done PTAL

**Comment by lsm5** [comment]
@flouthoc is this good to go?

**Comment by flouthoc** [comment]
LGTM @Luap99 @mheon @baude PTAL

**Pull request merged**